src/View/Helper/FormHelper.php

```diff
@@ -1649,7 +1649,7 @@ public function postLink($title, $url = null, array $options = [])
 
         $formName = str_replace('.', '', uniqid('post_', true));
         $formOptions = [
-            'action' => $this->Url->build($url),
+            'action' => Router::url($url),
```

Comment:
URLs used in HTML have to be HTML escaped, which is why we're using URL->build() here. Not escaping the `&` in urls will result in HTML validation errors.